experimental/generation/generator/packages/library/src/substitutions.ts

```diff
@@ -66,30 +71,37 @@ function substitutions(path: string, bindings: any, copies?: number, seed?: stri
     if (!seed) seed = '0'
 
     // Normalize bindings into a simple array and setup variables
-    let state = new Map<string, Variable>();
-    for (let binding of Object.keys(bindings)) {
-        let value = bindings[binding]
+    const state = new Map<string, Variable>();
+    for (const binding of Object.keys(bindings)) {
+        const value = bindings[binding]
         if (Array.isArray(value)) {
             state.set(binding, {index: -1, values: (value as any).flat()})
-        } else {
+        } else if (typeof value === 'string' || typeof value === 'number') {
             state.set(binding, {index: -1, values: value ? [value] : []})
         }
     }
 
-    let result: string[] = []
-    let rand = random(seed)
-    let file = fs.readFileSync(path, 'utf8')
+    // Track all generated utterances
+    if (!('utterances' in bindings)) {
```

Comment:
Depending on your Typescript version this can be written a bit more succinctly as `bindings.utterances ??= new Set<string>();`

Reply:
I dont think this is available in TS yet. Maybe C#

Reply:
Looks like it's been available since 4.0.x, [see playground](https://www.typescriptlang.org/play?ts=4.0.5#code/DYUwLgBAZg9jBcEB2BXAtgIxAJwNwFgAoWGCAfjIF4IBGABlyA).